### DISCUSSION TITLE
Litestar and msgspec as alternatives to FastAPI and Pydantic?

### DISCUSSION THREAD

**Discussion by NormanShelley** [zauberzeug/nicegui#1351] [Ideas / Feature Requests]
I'm pleased that **NiceGUI** has picked up and enhanced the **JustPy** code base and continues to do so. Both are lovely and excellent pieces of work. Kudos to all for their hard work and effort.

Recently I came across **msgspec** and then **Litestar** (which just started including **msgspec**).
I think these might improve and enhance **NiceGUI** if they were added as either an alternative or replacement for **FastAPI** and **Pydantic**.

https://medium.com/@v3ss0n/litestar-2-0-a-faster-proper-fastapi-alternative-is-launching-soon-cf543a0931f8
https://github.com/litestar-org/litestar
https://gist.github.com/jcrist/782c463f9eb52211e255caa96f57ab97
https://www.libhunt.com/compare-msgspec-vs-pydantic
https://jcristharif.com/msgspec/benchmarks.html

**Comment by rodja** [comment]
Litestar (formerly known as Starlite) is definitely an interesting alternative to FastAPI. The question [has been asked last November inside another discussion](https://github.com/zauberzeug/nicegui/discussions/135#discussioncomment-4030080). It's a good time to revisit and update my opinion:

- FastAPI is still much more well-known (see [star-history](https://star-history.com/#litestar-org/litestar&tiangolo/fastapi&Timeline) and  [Google Trends](https://trends.google.com/trends/explore?date=now%201-d&geo=DE&q=fastapi,litestar&hl=de))
- FastAPI is less opinionated than Starlite which makes it easier for us to implement our ideas we have with NiceGUI (for example storage)
- we have gained a lot of experience with FastAPI over the last years and are happy with it's reliability -- but can not really judge because we have not done proper comparisons
- reworking the code to use Litestar instead of FastAPI would be a very huge undertaking (see https://github.com/zauberzeug/nicegui/discussions/528); we should be absolutely sure that is what NiceGUI needs right now

Please do not get discouraged by this list. We are open for discussion and pro/cons. It's just a really big change which we should take very serious.

**Reply by rodja** [reply]
If FastAPI really stalls in the future we are very confident that we could change to another backend engine like Litestar.
